src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -560,8 +562,18 @@ export function registerTerminalActions() {
 				const endOfLinePreference = isWindows ? EndOfLinePreference.LF : EndOfLinePreference.CRLF;
 				text = editor.getModel().getValueInRange(selection, endOfLinePreference);
 			}
-			instance.sendText(text, true, true);
+			await instance.sendText(text, true, true);
 			await c.service.revealActiveTerminal();
+			const focusAfterRun = configurationService.getValue(TerminalSettingId.FocusAfterRun);
+			const focusTerminal = focusAfterRun === 'terminal' || (focusAfterRun === 'auto' && accessibilityService.isScreenReaderOptimized());
+			if (focusTerminal) {
+				instance.focus(true);
+			} else if (focusAfterRun === 'accessible-buffer') {
+				const contribution = instance.getContribution('terminal.accessible-buffer');
+				if (contribution) {
+					(contribution as any).show();
```

Comment:
I could add a method to `ITerminalInstance` to focus the accessible buffer instead of this hack? other ideas @tyriar?

Reply:
The right way to do this imo is to have a request focus event and listen/react to that conditionally in the contribution code; enabling features to be implemented in the terminalContrib with minimal changes in core.

Reply:
took a different approach, lmk what you think.